website/docs/r/nsxv_firewall.markdown

```diff
@@ -0,0 +1,255 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_nsxv_firewall"
+sidebar_current: "docs-vcd-resource-nsxv-firewall"
+description: |-
+  Provides a vCloud Director firewall resource for advanced edge gateways (NSX-V). This can be used
+  to create, modify, and delete firewall rules.
+---
+
+# vcd\_nsxv\_firewall
+
+Provides a vCloud Director firewall resource for advanced edge gateways (NSX-V). This can be used to
+create, modify, and delete firewall rules. Replaces
+[`vcd_firewall_rules`](/docs/providers/vcd/r/firewall_rules.html) resource.
+
+~> **Note:** This resource requires advanced edge gateway. For non-advanced edge gateways please
+use the [`vcd_firewall_rules`](/docs/providers/vcd/r/firewall_rules.html) resource.
+
+## Example Usage 1 (Minimal input)
+
+```hcl
+resource "vcd_nsxv_firewall" "my-rule-1" {
+  org          = "my-org"
+  vdc          = "my-vdc"
+  edge_gateway = "my-edge-gateway"
+
+  source {
+    ip_addresses = ["any"]
+  }
+
+  destination {
+    ip_addresses = ["192.168.1.110"]
+  }
+
+  service {
+    protocol = "any"
+  }
+}
+```
+
+
+## Example Usage 2 (Multiple services)
+
+```hcl
+resource "vcd_nsxv_firewall" "my-rule-1" {
+  org          = "my-org"
+  vdc          = "my-vdc"
+  edge_gateway = "my-edge-gateway"
+
+  source {
+    ip_addresses       = ["any"]
+    gateway_interfaces = ["internal"]
+  }
+
+  destination {
+    ip_addresses = ["192.168.1.110"]
+  }
+
+  service {
+    protocol = "icmp"
+  }
+
+  service {
+    protocol = "tcp"
+    port     = "443"
+  }
+}
+```
+
+## Example Usage 3 (Use exclusion in source)
```

Comment:
Given that exclusion is described in separate section and that I just requested one more example, I suggest moving exclusions examples to the mentioned exclusion section below.

Reply:
I remember I had a comment but this got lost. Given that the `exclude` is a single attribute in the `source`/`destination`blocks I think it would be too much to insert an example near it.  I have added a link in to this example in the `exclude`attributes note. Is it any better?

Reply:
Yes, neat!